src/unlock/admin_list.rs

```diff
@@ -0,0 +1,250 @@
+use lazy_static::lazy_static;
+use sparse_merkle_tree::default_store::DefaultStore;
+
+use sparse_merkle_tree::{SparseMerkleTree, H256};
+
+use crate::types::xudt_rce_mol::RCDataBuilder;
+use crate::types::xudt_rce_mol::{
+    RCDataUnion, RCRuleBuilder, SmtProofBuilder, SmtProofEntryBuilder, SmtProofEntryVec,
+    SmtProofEntryVecBuilder,
+};
+use bytes::Bytes;
+use ckb_hash::{Blake2b, Blake2bBuilder};
+use ckb_types::molecule;
+use ckb_types::prelude::*;
+use sparse_merkle_tree::traits::Hasher;
+
+lazy_static! {
+    static ref SMT_EXISTING: H256 = H256::from([
+        1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+        0, 0,
+    ]);
+    static ref SMT_NOT_EXISTING: H256 = H256::from([
+        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+        0, 0,
+    ]);
+}
+
+type SMT = SparseMerkleTree<CKBBlake2bHasher, H256, DefaultStore<H256>>;
+
+// on(1): white list
+// off(0): black list
+const WHITE_BLACK_LIST_MASK: u8 = 0x2;
+
+// on(1): emergency halt mode
+// off(0): not int emergency halt mode
+const EMERGENCY_HALT_MODE_MASK: u8 = 0x1;
+const BLAKE2B_KEY: &[u8] = &[];
+const BLAKE2B_LEN: usize = 32;
+const PERSONALIZATION: &[u8] = b"ckb-default-hash";
+struct CKBBlake2bHasher(Blake2b);
+
+impl Default for CKBBlake2bHasher {
+    fn default() -> Self {
+        let blake2b = Blake2bBuilder::new(BLAKE2B_LEN)
+            .personal(PERSONALIZATION)
+            .key(BLAKE2B_KEY)
+            .build();
+        CKBBlake2bHasher(blake2b)
+    }
+}
+
+impl Hasher for CKBBlake2bHasher {
+    fn write_h256(&mut self, h: &H256) {
+        self.0.update(h.as_slice());
+    }
+    fn finish(self) -> H256 {
+        let mut hash = [0u8; 32];
+        self.0.finalize(&mut hash);
+        hash.into()
+    }
+    fn write_byte(&mut self, b: u8) {
+        self.0.update(&[b][..]);
+    }
+}
+
+fn new_smt(pairs: Vec<(H256, H256)>) -> SMT {
+    let mut smt = SMT::default();
+    for (key, value) in pairs {
+        smt.update(key, value).unwrap();
+    }
+    smt
+}
+
+// return smt root and proof
+pub fn build_smt_on_wl(hashes: &[[u8; 32]], on: bool) -> (H256, Vec<u8>) {
```

Comment:
Use `ckb_types::H256` instead of `[u8; 32]`, and rename `sparse_merkle_tree::H256` as `SmtH256`